Source: Written by Michael Forney
Upstream: http://www.spinics.net/lists/hotplug/msg04142.html
Reason: Fix systemdsystemunitdir configure option

From a275e7a5510ecd0db977c30f5de822437a886998 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Thu, 26 Aug 2010 08:42:25 -0700
Subject: [PATCH] Don't install systemd scripts with --without-systemdsystemunitdir

Currently, the scripts get installed to /no/ if that option is
specified.
---
 configure.ac |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0a18bb6..31c5b18 100644
--- a/configure.ac
+++ b/configure.ac
@@ -66,8 +66,10 @@ AC_SUBST([FIRMWARE_PATH], [$FIRMWARE_PATH])
 AC_ARG_WITH([systemdsystemunitdir],
 	AS_HELP_STRING([--with-systemdsystemunitdir=DIR], [Directory for systemd service files]),
 	[], [with_systemdsystemunitdir=$($PKG_CONFIG --variable=systemdsystemunitdir systemd)])
-AC_SUBST([systemdsystemunitdir], [$with_systemdsystemunitdir])
-AM_CONDITIONAL(WITH_SYSTEMD, [test -n "$with_systemdsystemunitdir"])
+if test "x$with_systemdsystemunitdir" != xno; then
+    AC_SUBST([systemdsystemunitdir], [$with_systemdsystemunitdir])
+fi
+AM_CONDITIONAL(WITH_SYSTEMD, [test -n "$with_systemdsystemunitdir" -a "x$with_systemdsystemunitdir" != xno ])
 
 AC_ARG_ENABLE([extras],
 	AS_HELP_STRING([--disable-extras], [disable extras with external dependencies]),
-- 
1.7.2.2

